src/Pact/Types/Term.hs

```diff
@@ -608,41 +754,83 @@ data Term n =
     , _tMeta :: !Meta
     , _tInfo :: !Info
     }
-    deriving (Functor,Foldable,Traversable,Eq)
-
-instance Show n => Show (Term n) where
-    show TModule {..} =
-      "(TModule " ++ show _tModuleDef ++ " " ++ show (unscope _tModuleBody) ++ ")"
-    show (TList bs _ _) = "[" ++ unwords (map show bs) ++ "]"
-    show TDef {..} = show _tDef
-    show TNative {..} =
-      "(TNative " ++ asString' _tNativeName ++ " " ++ showFunTypes _tFunTypes ++ " " ++ unpack _tNativeDocs ++ ")"
-    show TConst {..} =
-      "(TConst " ++ asString' _tModule ++ "." ++ show _tConstArg ++ " " ++ show _tMeta ++ ")"
-    show (TApp a _) = show a
-    show (TVar n _) = "(TVar " ++ show n ++ ")"
-    show (TBinding bs b c _) = "(TBinding " ++ show bs ++ " " ++ show (unscope b) ++ " " ++ show c ++ ")"
-    show (TObject bs _ _) =
-      "{" ++ intercalate ", " (map (\(a,b) -> show a ++ ": " ++ show b) bs) ++ "}"
-    show (TLiteral l _) = show l
-    show (TGuard k _) = show k
-    show (TUse u _) = show u
-    show (TValue v _) = BSL.toString $ encode v
-    show (TStep ent e r _) =
-      "(TStep " ++ show ent ++ " " ++ show e ++ maybeDelim " " r ++ ")"
-    show TSchema {..} =
-      "(TSchema " ++ asString' _tModule ++ "." ++ asString' _tSchemaName ++ " " ++
-      show _tFields ++ " " ++ show _tMeta ++ ")"
-    show TTable {..} =
-      "(TTable " ++ asString' _tModule ++ "." ++ asString' _tTableName ++ ":" ++ show _tTableType
-      ++ " " ++ show _tMeta ++ ")"
+    deriving (Functor,Foldable,Traversable,Eq,Show)
+
+-- | Newtype to avoid introducing an orphan instance for @Pretty Var@. Used
+-- just for pretty printing (@prettyUnscope@).
+newtype PrettyVar a b = PrettyVar (Var a b)
+
+instance (Pretty a, Pretty b) => Pretty (PrettyVar a b) where
+  pretty (PrettyVar (B b)) = pretty b
+  pretty (PrettyVar (F a)) = pretty a
+
+prettyUnscope
+  :: (Functor f, Pretty (f (PrettyVar b (f a))))
+  => Scope b f a -> Doc
+prettyUnscope = pretty . fmap PrettyVar . unscope
+
+instance Pretty n => Pretty (Term n) where
+  pretty = \case
+    TModule{..} -> parensSep
+      [ "TModule", pretty _tModuleDef, prettyUnscope _tModuleBody ]
+    TList{..} -> prettyList _tList
+    TDef{..} -> pretty _tDef
+    TNative{..} -> dullgreen ("native function `" <> pretty (asString' _tNativeName) <> "`")
```

Comment:
This should be capitalized `Native function ...` in keeping with the convention you've set

Reply:
I don't consider this the capitalization convention. Considering things that are capitalized:

* `TModule`, `TBinding`, `TStep`, `TSchema`, `TTable`: All `Term` constructors that I'm basically punting on by following a format similar to their previous `show` instance. I'm not sure where, if ever, `pretty` can be called on these.
* `KeySet`, `PactGuard`, `ModuleGuard`, `UserGuard`: Same story for these non-`Term`s.

In both cases I'd love to know (1) if there's a different format we might want to use, and (2) how I can trigger these (and test them).

Reply:
See above, I'd like to go with a lispy/sexpy presentation with lowercase, and defer to the inner datastructure on TModule etc. With native I might suggest just `(native fun-name ...)`